cmd/integration_link.go

```diff
@@ -253,13 +280,30 @@ List of available integrations:
 				errorQuit(errors.New("cannot define both destroy-on-stale and no-destroy-on-stale"))
 			}
 
-			currentApp := detect.CurrentApp(c)
-			params, err := integrationlink.CheckAndFillParams(c, currentApp)
-			if err != nil {
-				errorQuit(err)
+			allowReviewAppsFromForks := c.Bool("allow-review-apps-from-forks")
+			noAllowReviewAppsFromForks := c.Bool("no-allow-review-apps-from-forks")
+
+			if allowReviewAppsFromForks && noAllowReviewAppsFromForks {
+				errorQuit(errors.New("cannot define both allow-review-apps-from-forks and no-allow-review-apps-from-forks"))
+			}
+
+			awareOfSecurityRisks := c.Bool("aware-of-security-risks")
+
+			if allowReviewAppsFromForks && !awareOfSecurityRisks {
+				stillAllowed, err := askForConfirmation(reviewAppsFromForksSecurityWarning)
+				if err != nil {
+					errorQuit(err)
+				}
+				err = c.Set("allow-review-apps-from-forks", strconv.FormatBool(stillAllowed))
```

Comment:
That way we actually allow review apps from forks even if the user answered "No".

This is because `CheckAndFillParams` only checks if `allow-review-apps-from-forks` is set, not its actual value.

I would suggest to unset `allow-review-apps-from-forks` if `stillAllowed` is false.

Reply:
I am still facing the exact same issue

Did you submit a fix for that one?

To reproduce:

```console
$ ./scalingo-cli -a test-app integration-link
Application: test-app (63691985884fef00111d2a6b)
Integration: GitHub (22a2daa4-b636-41c7-8b1b-9cd10e549fa5)
Linker: aurelien-reeves-scalingo

Repository: aurelien-reeves-scalingo/review-apps-securisation
Auto Deploy: ✘
Review Apps Deploy: ✔
	Destroy on Close: instantly
	Destroy on Stale: ✘
	Automatic creation from forks: ✘

$ ./scalingo-cli -a test-app integration-link-update --allow-review-apps-from-forks
  /!\  Only allow automatic review apps deployments from forks if you trust the owners of those forks, as this could lead to security issues. More info here: https://doc.scalingo.com/platform/app/review-apps#addons-collaborators-and-environment-variables
? Are your sure? No
-----> Your app 'test-app' integration link has been updated.

$ ./scalingo-cli -a test-app integration-link
Application: test-app (63691985884fef00111d2a6b)
Integration: GitHub (22a2daa4-b636-41c7-8b1b-9cd10e549fa5)
Linker: aurelien-reeves-scalingo

Repository: aurelien-reeves-scalingo/review-apps-securisation
Auto Deploy: ✘
Review Apps Deploy: ✔
	Destroy on Close: instantly
	Destroy on Stale: ✘
	Automatic creation from forks: ✔

```

We can see that the answer "No" to the update request did not prevent the update to be done